tests/unit/validation/test_validation_functions.py

```diff
@@ -431,6 +434,140 @@ def test_validate_values__infinite_sigmas(sensor_type, parameter):
+            [InvalidIdError, NotUniqueError, MultiFieldValidationError],
+        ),
+        (
+            "asym_power_sensor",
+            ["p_sigma", "q_sigma"],
+            [[0.1, NaN, NaN], [0.1, NaN, NaN]],
+            [InvalidIdError, NotUniqueError, MultiFieldValidationError],
+        ),
+        (
+            "asym_power_sensor",
+            ["p_sigma", "q_sigma"],
+            [[0.1, 0.1, 0.1], [0.1, 0.1, 0.1]],
+            [InvalidIdError, NotUniqueError],
+        ),
+    ],
+)
+def test_validate_values__bad_p_q_sigma(sensor_type, parameter, values, error_types):
+    def arbitrary_fill(array, sensor_type, parameter, values):
+        if sensor_type == "sym_power_sensor":
+            array[parameter[0]] = values[0]
+            array[parameter[1]] = values[1]
+        else:
+            array[parameter[0]][0][0] = values[0][0]
+            array[parameter[0]][0][1] = values[0][1]
+            array[parameter[0]][0][2] = values[0][2]
+            array[parameter[1]][0][0] = values[1][0]
+            array[parameter[1]][0][1] = values[1][1]
+            array[parameter[1]][0][2] = values[1][2]
+
+    sensor_array = initialize_array("input", sensor_type, 3)
+    arbitrary_fill(sensor_array, sensor_type, parameter, values)
+    all_errors = validate_values({sensor_type: sensor_array})
+
+    for error in all_errors:
+        assert any(isinstance(error, error_type) for error_type in error_types)
+        assert sensor_array["id"][0] == error.ids[0]
+
+
+@pytest.mark.parametrize(
+    ("values", "error_types"),
+    [
+        ([[NaN, NaN], [[NaN, NaN, NaN], [NaN, NaN, NaN]]], [InvalidIdError]),
+        ([[0.1, NaN], [[NaN, NaN, NaN], [NaN, NaN, NaN]]], [InvalidIdError, MultiFieldValidationError]),
+        ([[NaN, NaN], [[NaN, 0.1, NaN], [NaN, NaN, NaN]]], [InvalidIdError, MultiFieldValidationError]),
+        ([[NaN, NaN], [[NaN, 0.1, NaN], [NaN, 0.1, NaN]]], [InvalidIdError, MultiFieldValidationError]),
+        ([[0.1, 0.1], [[NaN, NaN, NaN], [NaN, NaN, NaN]]], [InvalidIdError, MultiFieldValidationError]),
+        ([[0.1, 0.1], [[NaN, NaN, NaN], [NaN, NaN, NaN]]], [InvalidIdError]),
+        ([[NaN, NaN], [[0.1, 0.1, 0.1], [0.1, 0.1, 0.1]]], [InvalidIdError]),
+        ([[0.1, 0.1], [[0.1, 0.1, 0.1], [0.1, 0.1, 0.1]]], [InvalidIdError]),
+    ],
+)
+def test_validate_values__bad_p_q_sigma_both_components(values, error_types):
+    def two_component_data(values):
+        node = initialize_array("input", "node", 1)
+        node["id"] = 123
+        sym_power_sensor = initialize_array("input", "sym_power_sensor", 1)
+        sym_power_sensor["p_sigma"] = values[0][0]
+        sym_power_sensor["q_sigma"] = values[0][1]
+        sym_power_sensor["id"] = 456
+        asym_power_sensor = initialize_array("input", "asym_power_sensor", 1)
+        asym_power_sensor["p_measured"] = values[1][0]
+        asym_power_sensor["q_measured"] = values[1][1]
+        asym_power_sensor["id"] = 789
```

Comment:
i actually meant 2 different components of the same type, e.g.

```suggestion
        sym_power_sensor = initialize_array("input", "sym_power_sensor", 2)
        sym_power_sensor["p_sigma"] = values[0][0]
        sym_power_sensor["q_sigma"] = values[0][1]
        sym_power_sensor["id"] = [456, 789]
```

such that e.g. the first one is OK but the second one is invalid (or vice versa). the error should contain the correct `id` of only the invalid one